spine/model/update.go

```diff
@@ -0,0 +1,64 @@
+package model
+
+import "github.com/DerAndereAndi/eebus-go/util"
+
+type Updater interface {
+	DoUpdate()
+}
+
+type UpdaterFactory[T any] interface {
+	NewUpdater(s *T, filterPartial *FilterType, filterDelete *FilterType) Updater
+}
+
+type UpdateDataProvider[T util.HashKeyer] interface {
+	ExistingData() []T
+	NewData() []T
+	UpdateSelektorHashKey() *string
+	DeleteSelektorHashKey() *string
+	HasIdentifier(*T) bool
+	CopyData(source *T, dest *T)
+}
+
+func UpdateList[T util.HashKeyer](dataProvider UpdateDataProvider[T]) []T {
+	newData := dataProvider.NewData()
+	existingData := dataProvider.ExistingData()
+	if len(newData) == 0 {
+		return existingData
+	}
+
+	// TODO: consider filterDelete
+	// TODO: Check if only single fields should be considered here
+
+	// check if selector is used
+	updateSelectorHashKey := dataProvider.UpdateSelektorHashKey()
+	if updateSelectorHashKey != nil {
+		return copyToSelectedData(dataProvider, &newData[0], *updateSelectorHashKey)
+	}
+
+	// check if items have no identifiers
+	if !dataProvider.HasIdentifier(&newData[0]) {
+		// no identifiers specified --> copy data to all existing items
```

Comment:
is this referenced in the Spec somewhere? if so, it might be good to add that reference (spec, chapter, ...)

Reply:
Comment added